use-timescale/extensions/pgvector.md

```diff
@@ -0,0 +1,478 @@
+---
+title: pgvector PostgreSQL extension
+excerpt: Use the pgvector extension with your Timescale service
+products: [cloud]
+keywords: [services, settings, extensions, pgvector]
+tags: [extensions, pgvector]
+---
+
+# The `pgvector` extension
+
+The `pgvector` PostgreSQL extension helps you to store and search over machine
+learning-generated embeddings. It provides different capabilities that allows
+you to identify both exact and approximate nearest neighbors. It is designed to
+work seamlessly with other PostgreSQL features, including indexing and querying.
+
+For more information about these functions and the options available, see the
+[pgvector][pgvector-repo] repository.
+
+## Use the `pgvector` extension to create a `chatbot`
+
+The `pgvector` PostgreSQL extension allows you to create, store, and query
+OpenAI [vector embeddings][vector-embeddings] in Timescale. Learn how to use
+[Retrieval Augmented Generation (RAG)][rag-docs] to create a `chatbot` that combines
+your data with the power of ChatGPT using OpenAI and pgvector. RAG addresses the
+problem that a foundational model such as GPT-3 or GPT-4 may be missing some
+information needed to give a good answer because that information was not in the
+dataset used to train the model. For example, the information is stored in
+private documents or only became available recently.
+
+### Prerequisites
+
+Before you begin, make sure you have:
+
+*   Installed Python.
+*   Created a [Timescale][cloud-login] service.
+*   Downloaded the cheatsheet when you created the service. This sheet contains
+    the connection details for the database you want to use as a vector database.
+*   Signed up for an [OpenAI developer account][openai-signup].
+*   Created and API key and made a note of your OpenAI [API key][api-key].
+    <Highlight type="note">
+    If you are on a free plan there may be rate limiting for your API requests.
+    </Highlight>
+*   Clone the [Timescale pgvector repository][timescale-pgvector] repository.
+
+<Procedure>
+
+### Using the `pgvector` extension to create a chatbot
+
+In this example, you create embeddings, insert the embeddings in Timescale and
+query the embeddings in Timescale using `pgvector`. The content for the
+embeddings is from the Timescale blog, specifically from the
+[Developer Q&A][developer-qa] section, which features posts by Timescale users talking
+about their real-world use cases.
```

Comment:
This section should be under the L2 heading, not the L3 heading.

Reply:
Done!!!